app/views/docs/configuration.phtml

```diff
@@ -22,10 +22,143 @@
 <h2><a href="#git" id="git">Configure GitHub App</a></h2>
 <p>
     Appwrite supports automatic deployments through Git integration. 
-    In order for Appwrite to access your repos, you must configure a GitHub app to enable this integration.
+    In order for Appwrite to access your repos, you must <a href="https://docs.github.com/en/apps/creating-github-apps/about-creating-github-apps/about-creating-github-apps" target="_blank" rel="noopener">create a GitHub app</a> to enable this integration.
+    The GitHub app requires the following configurations.
 </p>
 
-<p>You'll have to configure the following environment variables.</p>
+<p>
+    For automatic Git deployment to work, Appwrite needs to receive communication from GitHub, this means your Appwrite project must be accessible on the internet.
+    If you're running on <code>localhost</code>, you need to run a proxy like <a href="https://ngrok.com/" target="_blank" rel="noopener">ngrok</a>.
+</p>
+
+<h3><a href="#callback" id="callback">Callback URL</a></h3>
+<p>
+    GitHub will use <b>callback URLs</b> to redirect users back to Appwrite. 
+    Set these callback URLs under <b>Identifying and authorizing users</b>.
+</p>
+<table class="full">
+    <thead>
+        <tr>
+            <td>URLs</td>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>http://[HOSTNAME_OR_IP]/v1/vcs/github/callback</code></td>
+        </tr>
+        <tr>
+            <td><code>http://[HOSTNAME_OR_IP]/v1/account/sessions/oauth2/callback/github/console</code></td>
+        </tr>
+    </tbody>
+</table>
+
+<h3><a href="#post-install" id="post-install">Post Installation</a></h3>
+<p>
+    Check the <b>Redirect on update</b> box under the <b>Post installation</b> section.
+</p>
+
+
+<h3><a href="#github-webhooks" id="github-webhooks">Webhook</a></h3>
+<p>
+    GitHub will notify Appwrite about events like new commits using webhooks.
+    Under <b>Webhook</b>, you need to check the <b>Active</b> checkbox.
+    You also need to set the <b>Webhook URL</b> as <code>https://[HOSTNAME_OR_IP]/v1/vcs/github/events</code>.
+</p>
+
+<p>
+    If you're running Appwrite on <code>localhost</code>, GitHub can't send requests to Appwrite through webhooks and automatic deployments won't work.
+    You'll need host Appwrite on a server or use a proxy like <a href="https://ngrok.com/" target="_blank" rel="noopener">ngrok</a> to make Appwrite accessible to GitHub.
+</p>
+
+<h3><a href="#repository-permission" id="repository-permission">Repository Permissions</a></h3>
+<p>
+    Configure these permissions under the <b>Repository permission</b> dropdown.
+</p>
+<table class="full">
+    <thead>
+        <tr>
+            <td>Permission</td>
+            <td>Access</td>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>Administration</td>
+            <td>Read and write</td>
```

Comment:
weren't we going to explain why we need each of these permissions?

Reply:
We didn't have time

Reply:
Maybe in the future, when I can pull Khushboo and matej aside.